backEnd/index.js

```diff
@@ -42,114 +16,106 @@ app.listen(port, () => {
   console.log(`Server listening on the port::${port}`);
 });
 
-
 //below are the API methods, matched with apiServices in the Front End
 //NOTE: requires firebase integration, uses "users" variable for testing
 
 // Get all users for searching. Client will filter the array that is returned
 // sends list of users up to the client
-app.get('/api/users', (req, res) => {
-  console.log("get all users")
-  //todo: Get users from Firebase
-  res.json(users);
+app.get('/api/users', async (req, res) => {
+  try{
+    console.log("get all users")
+    let users = await userRepository.getAllUsers();
+    res.json(users);
+  }
+  catch(error) {
+    console.log('error' + error);
+    res.status(400);
```

Comment:
404 if the requested user doesn't exist. 400 if the request validation fails. 500 if the server fails